### ISSUE TITLE
code -r opens a new VSCode instance every time

### ISSUE THREAD

**Issue by theparham** [microsoft/vscode#207594]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.87.2
- OS Version: Windows 10 22H2 19045.4046

I'm having a weird issue trying to open my files from terminal in VSCode on windows. As far as I know using -r should force opening a file in the same instance of VSCode, but no matter how my settings is setup I still get a new instance opening up every time!

Already googled solutions and searched through stackoverflow, tried "window.openFilesInNewWindow": "off" and "window.openWithoutArgumentsInNewWindow": "off", no effects. I'm assuming this to be a bug, or am I missing something?

- note: working on a Django project repo if it helps.

Steps to Reproduce:

1. Open a new terminal (PowerShell) in your workspace.
2. cd to your desired directory
3. Run code -r yourfile.py
4. A new instance of VSCode will pop up opening yourfile.py instead of yourfile.py opening in the same VSCode instance

**Comment by theparham** [comment]
Additional comments from someone trying to help me on stackoverflow:

Q: Is the behavior the same whether you're using Terminal or just a Command-Prompt console window? 
A: Haven't checked it before, just tried and the result is the same. However just found out right after it opens a new instance, if I try to open any other files it actually uses the newly opened VSCode instance! but the main one where I'm at my workspace doesn't grab the file!

**Comment by theparham** [comment]
Found the culprit!

In case anyone else happen to have the same issue:

For some weird reason, if I run the executable from the command line, it would grab `code -r filename.ext` just fine. But if I open the executable using windows explorer it somehow gives me some sort of a local instance that terminal doesn't grab, even though both executables are running under the same username and privileges. Can't figure out why and how, but at least I've got a decent workaround for the moment thanks to [this guy](https://stackoverflow.com/users/3605497/tripp-kinetics) who's pointed me in the right direction on [stackoverflow](https://stackoverflow.com/questions/78155350/how-to-open-a-file-in-the-same-vscode-instance-using-terminal) :)
